From 262d2ab4dac1447a42397ef037e57ff862d58f47 Mon Sep 17 00:00:00 2001
From: Vivek Goyal <vgoyal@redhat.com>
Date: Fri, 5 Sep 2014 18:55:48 -0400
Subject: [kernel] kexec: use vscnprintf() instead of vsnprintf() in vmcoreinfo_append_str()

Message-id: <1409943373-12669-14-git-send-email-vgoyal@redhat.com>
Patchwork-id: 91524
O-Subject: [RHEL7.1 PATCH 13/38] kernel/kexec.c: use vscnprintf() instead of vsnprintf() in vmcoreinfo_append_str()
Bugzilla: 852066
RH-Acked-by: Dave Anderson <anderson@redhat.com>
RH-Acked-by: Baoquan He <bhe@redhat.com>
RH-Acked-by: Dave Young <dyoung@redhat.com>
RH-Acked-by: Chao WANG <chaowang@redhat.com>

BZ:852066

commit a19428e5c3f4f39c302cb97b847dfd23613145c1
Author: Chen Gang <gang.chen@asianux.com>
Date: Mon, 27 Jan 2014 17:07:13 -0800

vsnprintf() may let 'r' larger than sizeof(buf), in this case, if 'r' is
also less than "vmcoreinfo_max_size - vmcoreinfo_size" (left size of
destination buffer), next memcpy() will read the unexpected addresses.

Signed-off-by: Chen Gang <gang.chen@asianux.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/kernel/kexec.c b/kernel/kexec.c
index bbb53b5..1aa71eb 100644
--- a/kernel/kexec.c
+++ b/kernel/kexec.c
@@ -1595,7 +1595,7 @@ void vmcoreinfo_append_str(const char *fmt, ...)
  size_t r;
 
  va_start(args, fmt);
- r = vsnprintf(buf, sizeof(buf), fmt, args);
+ r = vscnprintf(buf, sizeof(buf), fmt, args);
  va_end(args);
 
  r = min(r, vmcoreinfo_max_size - vmcoreinfo_size);
-- 
1.7.1